include/bluetooth/mesh/cfg_cli.h

```diff
@@ -716,13 +716,13 @@ int bt_mesh_cfg_hb_pub_get(u16_t net_idx, u16_t addr,
 
 /** @brief Get the current transmission timeout value.
  *
- *  @return The configured transmission timeout.
+ *  @return The configured transmission timeout in milliseconds.
  */
 s32_t bt_mesh_cfg_cli_timeout_get(void);
 
 /** @brief Set the transmission timeout value.
  *
- *  @param timeout The new transmission timeout.
+ *  @param timeout The new transmission timeout in milliseconds.
  */
 void bt_mesh_cfg_cli_timeout_set(s32_t timeout);
```

Comment:
I was looking at the way the timeout is used in these functions, and I think the most convenient approach would be to just change `get()` to return `k_timeout_t` and `set()` to take `k_timeout_t`. Same goes for the corresponding `bt_mesh_health_cli_*()` functions.

Reply:
Yeah, I was debating which one was best here. Changing to k_timeout_t is an API break, so question is if that is okey?
The advantage to having it as s32_t is that it can be compared, since there is no way to go from k_timeout_t back to milliseconds again. I'm not sure if that is needed or not, so I leave that you.

If I remove this part I think I can remove all of those compares against K_TICKS_FOREVER as well.

Reply:
You have an API break either way. Code that was passing e.g. `K_SECONDS(5)` to `set(s32_t timeout)` would break, whereas if we change to `k_timeout_t` it wouldn't break. Code storing the return value of `set()` will likely break with `k_timeout_t` however, however this isn't a particularly useful function and the usage in the shell is probably the only place.

Reply:
> The advantage to having it as s32_t is that it can be compared, since there is no way to go from k_timeout_t back to milliseconds again.

That's a fair point, but doesn't apply to these particular functions. If you need to compare for something else than equality trying to keep ms is probably most practical.

Reply:
Yes, I do agree that if we're talking APIs that we need to break anyway then we're better off using the new `k_timeout_t` type for future proofing.

Reply:
By returning `k_timeout_t` here the `get()` function is not very useful, as you can only compare against K_FOREVER and K_NO_WAIT.
I'm wondering if there should be an `s32_t k_timeout_msec_get(k_timeout_t);`

Reply:
Yes. Go with s32_t. That’s also consistent with kernel “get” APIs

Reply:
Without msec in the name, that is

Reply:
I'm thinking here of the likes of `k_uptime_get()` and `k_delayed_work_remaining_get()`

Reply:
@andyross @pabigot @nordic-krch 
Thoughts on creating a function:
`s32_t k_timeout_get(k_timeout_t t) { return k_tick_to_ms_floor(T); }`?

Reply:
So I misunderstood here, I thought @jhedberg  said yes to:
 > I'm wondering if there should be an s32_t k_timeout_msec_get(k_timeout_t);

But he meant having s32_t as return value on the `bt_mesh_cfg_cli_timeout_get()`, I have now updated to this. :)

Reply:
Ok. s32_t as input to set, and output to get. And internally using SYS_TIMEOUT_MS to handle SYS_FOREVER_MS